lib/at_cmd_parser/at_cmd_parser.c

```diff
@@ -343,3 +329,29 @@ int at_parser_max_params_from_str(const char *at_params_str,
 
 	return err;
 }
+
+enum at_cmd_type at_parser_cmd_type_get(const char *at_cmd)
+{
+	enum at_cmd_type type;
+
+	if (!is_command(at_cmd)) {
+		return AT_CMD_TYPE_UNKNOWN;
+	}
+
+	at_cmd += sizeof("AT+") - 1;
+
+	while (is_valid_notification_char(*at_cmd)) {
+		at_cmd++;
+	}
+
+	if ((*at_cmd == AT_CMD_SEPARATOR) &&
+	    (*(at_cmd + 1) == AT_CMD_READ_TEST_IDENTIFIER)) {
+		type = AT_CMD_TYPE_TEST_COMMAND;
+	} else if (*at_cmd == AT_CMD_READ_TEST_IDENTIFIER) {
+		type = AT_CMD_TYPE_READ_COMMAND;
+	} else {
+		type = AT_CMD_TYPE_SET_COMMAND;
```

Comment:
In case of SET command without any parameter, is "at_cmd" pointer moved to the end of the input string? In that case, can we still use * operator?

Reply:
The pointer is a local to this function, any modifications to it are not visible outside.

Reply:
Again need to call is_lfcr() to confirm the termination of AT string, before using * to access at_cmd pointer.